packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2709,8 +2710,13 @@ export class ContainerRuntime
 		const savedOp = (messageCopy.metadata as ISavedOpMetadata)?.savedOp;
 		const logLegacyCase = getSingleUseLegacyLogCallback(this.logger, messageCopy.type);
 
-		// We expect runtime messages to have JSON contents - deserialize it in place.
-		ensureContentsDeserialized(messageCopy, hasModernRuntimeMessageEnvelope, logLegacyCase);
+		let runtimeBatch: boolean =
+			hasModernRuntimeMessageEnvelope || isUnpackedRuntimeMessage(messageCopy);
+		if (runtimeBatch) {
+			// We expect runtime messages to have JSON contents - deserialize it in place.
```

Comment:
@vladsud -- Do you recall if "old" runtime messages (no "Op" envelope) had their contents stringified as well?  I'm assuming so but didn't go check and don't have a concrete memory of that time.

Reply:
Ok I traced `JSON.stringify` back almost 6 years, to a7cb2245c1aa99c3c818336ecdd63191259962ff, which is past the creation of `container.ts`, so I think we're good 😅  Also, counting on snapshot tests to run old trailing ops through the new code

Reply:
I think if you pass snapshot tests, you are good. I've started creating those long before Loop declared no-return point RE supporting files at rest

Reply:
That said, I think with really small exception very early in our journey (where I think we were sending objects and PUSH would serialize them), I think it was always the case that storage had strings and were returning strings when fetching ops. So, one question to ask here - shouldn't we always parse content irrespective if it's modern op or not?

Reply:
We do